### PR TITLE
Added support for $ifNull multi field support, $isNumber, $isArray

### PR DESCRIPTION
**Pull request by David-Taub**
Added support for $ifnull with multiple input fields, added in MongoDB v4.4
solves issue #748

### CODE REVIEW

File: tests/test__collection_api.py [hunk 1]
```diff
@@ -3787,6 +3787,20 @@ def test__aggregate_project_if_null(self):
...
+        actual = self.db.collection.aggregate([
+            {'$match': {'_id': 1}},
+            {'$project': collections.OrderedDict([
+                ('_id', False),
+                ('a_and_b', {'$ifNull': ['$elem_a', '$elem_b', '<missing_both>']}),
+                ('b_and_a', {'$ifNull': ['$elem_b', '$elem_a', '<missing_both>']}),
+                ('b_and_c', {'$ifNull': ['$elem_b', '$elem_c', '<missing_both>']}),
+            ])}
+        ])
+        self.assertEqual([{'a_and_b': '<present_a>', 'b_and_a': '<present_a>',
```
Comment (pcorpet) [inline tests/test__collection_api.py]: Please break after `self.assertEqual(`

**Review by pcorpet — CHANGES_REQUESTED** [review]
Thanks for the feature.

Please add a test in test__mongomock to make sure the feature stays aligned with pymongo + MongoDB behavior.

File: tests/test__collection_api.py (hunk 1 above)
Comment (David-Taub) [reply tests/test__collection_api.py]: Note that I added a helper `get_mongodb_version` that checks the version of MongoDB
I added the newer $ifNull syntax that is supported only on versions higher than v4.4 which is currently used in this project. 
When this project upgrades into MongoDB v5, the test will pass and the syntax can be added

Do you think the PR can be approved if the aggregation fails depending on the MongoDB version

**Comment by David-Taub** [comment]
@pcorpet, Note that I added a helper get_mongodb_version that checks the version of MongoDB
I added the newer $ifNull syntax that is supported only on versions higher than v4.4 which is currently used in this project.
When this project upgrades into MongoDB v5, the test will pass and the syntax will be used and not fail (added a check for MongoDB version for that)

File: tests/test__mongomock.py [hunk 2]
```diff
@@ -3000,6 +3000,86 @@ def test__aggregate_count(self):
...
+        self.cmp.compare.aggregate([
+            {
+                '$project': {
+                    'a': {'$ifNull': ['$elem_a', '<missing_a>']},
+                    'b': {'$ifNull': ['$elem_b', '<missing_b>']},
+                }
+            }
+        ])
+
+        @skipIf(
```
Comment (pcorpet) [inline tests/test__mongomock.py]: Please deindent otherwise this function is nested in the other one.

File: mongomock/aggregate.py [hunk 3]
```diff
@@ -156,6 +156,10 @@
     '$arrayToObject',
     '$objectToArray',
 ]
+type_operators = [
```
Comment (pcorpet) [inline mongomock/aggregate.py]: Please add those in a separate PR.

**Review by pcorpet — CHANGES_REQUESTED** [review]
Thanks for your tests that revealed that we need more work.

I think that you're right that we could use a mechanism to handle different server versions in here, but I'd rather not introduce code that is not tested in the CI.

I think I'm going to first switch to MongoDB v5 for testing so that you can rebase and use that for the tests. Please remove the new helper and the skipIf parts.

**Comment by pcorpet** [comment]
> Added support for $ifnull with multiple input fields, added in MongoDB v4.4 solves issue #748
Note that as opposed to what you wrote in your first comment here, the feature is only available *after* v4.4 not in 4.4.

**Comment by pcorpet** [comment]
OK, this is done now, you have a `mongomock.SERVER_VERSION` that you can use

File: tests/test__mongomock.py (hunk 2 above)
Comment (pcorpet) [reply tests/test__mongomock.py]: Done.

File: mongomock/aggregate.py (hunk 3 above)
Comment (pcorpet) [reply mongomock/aggregate.py]: Done in #789

**Pull request merged**